resolver/resolver.go

```diff
@@ -82,6 +82,14 @@ func exchangers(timeout time.Duration, protos ...string) map[string]exchanger.Ex
 	return exs
 }
 
+// Returns the currents set of enumerable data from the resolver
+// this data only makes sense to be read-only
+func (res *Resolver) enumerableData() records.EnumerationData {
+	res.rsLock.Lock()
```

Comment:
stop referring to res.rs directly, get rid of the lock. this entire func is then reduced to `return res.records().EnumData`